lib/authentic/validator.rb

```diff
@@ -65,5 +67,11 @@ def decode_jwt(token)
     rescue JSON::JWT::InvalidFormat
       raise InvalidToken, 'invalid JWT format'
     end
+
+    private
+
+    def verify_claims(jwt)
+      raise InvalidToken, 'expired JWT' unless Time.at(jwt[:exp]) > Time.now
```

Comment:
I don't think this needs to be in its own method as it's just a one liner. If we're going to extract a method, it would in my opinion make more sense to pull the entire body of that tap block out to a separate method instead.

At this point though, the method is so small that I don't even think it's worth the added cognitive overhead of a method call. Otherwise, looks totally fine to me as long as these expirations are all UTC timestamps. Up to you if you want to change the structure of the code.

Reply:
good point.  I started with the idea of needing to validate more claims, but at this point I think `exp` is all we need.  I've moved the check back into `ensure_valid` and moved the check above `jwt.verify!` since there is no need to do the verify step if the token is expired.

Reply:
Cool, that's a good optimization to make!